scripts/preprocess_spec.py

```diff
@@ -30,24 +39,67 @@ def _to_camel_case(s):
     return ''.join(_title(y) for y in s.split("_"))
 
 
-def iterate_through_operations(spec, func):
+def apply_funcs_to_spec(spec, func, *params):
```

Comment:
nit: Given that this function only applies a single func, consider renaming to ``appy_func_to_spec``